### PR TITLE
Documentation cleanup for the BayeuxClient library

### PR DESCRIPTION
**Pull request by ppetrosh**
Develop

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,178 +1,194 @@
...
 
 **To add this library to your project, add** `#require "BayeuxClient.agent.lib.nut:1.0.0"` **to the top of your agent code**
 
-## Library Usage ##
-
-## Bayeux.Client Class ##
+## Bayeux.Client Class Usage ##
 
-### Constructor: Bayeux.Client(*configuration[, onConnected[, onDisconnected]]*) ###
+### Constructor: Bayeux.Client(*configuration[, onConnected][, onDisconnected]*) ###
```
Comment (ragrus-nbl) [inline README.md:20]: @smittytone Are you sure it is right? We can't pass the *onDisconnected* param without the *onConnected*.

Comment (ElectricImpSampleCode) [reply README.md:20]: Current style is to bracket optional params separately. This is what we do everywhere else. You can pass a function into *onDisconnected* without (kind of) passing one into *onConnected*:

```
local i = Bayeux.Client(someTable, null, function(error) {
   server.log("onDisconnect handler called");
});
```

Comment (ragrus-nbl) [reply README.md:20]: Ok, thanks! Just wanted to make sure

File: README.md [hunk 2]
```diff
@@ -1,178 +1,194 @@
...
+This callback is called every time the client connects. The client is considered to be connected when the handshake and the first *connect* messages were successful. To learn what handshake and *connect* messages are, please see the [Bayeux protocol specification](https://docs.cometd.org/current/reference/#_bayeux).
 
-This is a good place to call the [connect()](#connect) method again if it was an unexpected disconnection.
+The callback has one parameter of its own:
 
-| Parameter | Data Type | Description |
+| Parameter | Type | Description |
 | --- | --- | --- |
-| *reason* | [Bayeux.Error](#bayeuxerror-class) | null if the disconnection was caused by the [disconnect()](#disconnect) method, error details which explains a reason of the disconnection otherwise. |
+| *error* | [Bayeux.Error](#bayeuxerror-class) | This will be `null` if the connection is successful, otherwise an error message string |
```
Comment (ragrus-nbl) [inline README.md:58]: @smittytone We pass in to the callback an instance of Bayeux.Error class, not a string. We do so for all callbacks in this library.

File: README.md [hunk 3]
```diff
@@ -1,178 +1,194 @@
...
+  }
 }
 
-config <- {
-    "url" : "yourBayeuxServer.com",
-    "requestHeaders": {
-        "Authorization" : "<Your authorization header if needed>"
-    }
+config <- { "url" : "yourBayeuxServer.com",
+            "requestHeaders": { "Authorization" : "YOUR_UTHORIZATION_HEADER"}
```
Comment (ragrus-nbl) [inline README.md:105]: `YOUR_UTHORIZATION_HEADER` contains a typo

**Comment by ppetrosh** [comment]
@smittytone @ragrus-nbl do we have all the issues/questions addressed at this point? Can we merge the PR and publish the library?

**Comment by ppetrosh** [comment]
Merging and I'll take care of the pending issues.

**Pull request merged**